EngineLayer/CommonParameters.cs

```diff
@@ -8,115 +8,146 @@ namespace EngineLayer
+        public double DeconvolutionIntensityRatio { get; private set; }
+        public int DeconvolutionMaxAssumedChargeState { get; private set; }
+        public Tolerance DeconvolutionMassTolerance { get; private set; }
+        public int TotalPartitions { get; private set; }
+        public bool BIons { get; private set; }
+        public bool YIons { get; private set; }
+        public bool ZdotIons { get; private set; }
+        public bool CIons { get; private set; }
+        public Tolerance ProductMassTolerance { get; private set; }
+        public Tolerance PrecursorMassTolerance { get; private set; }
+        public bool CompIons { get; private set; }
+        public double ScoreCutoff { get; private set; }
+        public DigestionParams DigestionParams { get; private set; }
+        public bool ReportAllAmbiguity { get; private set; }
+        public int TopNpeaks { get; private set; }
+        public double MinRatio { get; private set; }
+        public bool TrimMs1Peaks { get; private set; }
+        public bool TrimMsMsPeaks { get; private set; }
+        public bool UseDeltaScore { get; private set; }
+        public bool CalculateEValue { get; private set; }
 
         #endregion Public Properties
 
         #region Public Methods
 
         public CommonParameters Clone()
         {
-            return (CommonParameters)this.MemberwiseClone();
+            return new CommonParameters(
+                BIons: this.BIons,
+                YIons: this.YIons,
+                ZdotIons: this.ZdotIons,
+                CIons: this.CIons,
+                DoPrecursorDeconvolution: this.DoPrecursorDeconvolution,
+                UseProvidedPrecursorInfo: this.UseProvidedPrecursorInfo,
+                DeconvolutionIntensityRatio: this.DeconvolutionIntensityRatio,
+                DeconvolutionMaxAssumedChargeState: this.DeconvolutionMaxAssumedChargeState,
+                ReportAllAmbiguity: this.ReportAllAmbiguity,
+                CompIons: this.CompIons,
+                TotalPartitions: this.TotalPartitions,
+                ScoreCutoff: this.ScoreCutoff,
+                TopNpeaks: this.TopNpeaks,
+                MinRatio: this.MinRatio,
+                TrimMs1Peaks: this.TrimMs1Peaks,
+                TrimMsMsPeaks: this.TrimMsMsPeaks,
+                UseDeltaScore: this.UseDeltaScore,
+                CalculateEValue: this.CalculateEValue,
+                ProductMassTolerance: this.ProductMassTolerance,
+                PrecursorMassTolerance: this.PrecursorMassTolerance,
+                DeconvolutionMassTolerance: this.DeconvolutionMassTolerance,
+                MaxThreadsToUsePerFile: this.MaxThreadsToUsePerFile,
+                DigestionParams: this.DigestionParams,
+                ListOfModsVariable: this.ListOfModsVariable,
+                ListOfModsFixed: this.ListOfModsFixed
+            );
+        }
+
+        public void SetTaskDescriptor(string TaskDescriptor)
```

Comment:
This set method is unnecessary since the variable still has a public set.

Reply:
You could just leave the others two as public sets and remove the other to set methods, too.